src/vehicle.cpp

```diff
@@ -3756,9 +3764,35 @@ void vehicle::idle(bool on_map) {
 
     if( on_map ) {
         update_time();
+        if(scoop_on){
+            operate_scoop();
+        }
+    }
+}
+void vehicle::operate_scoop(){
+    std::vector<int> scoops=all_parts_with_feature("SCOOP");
+    auto c=get_points();
+    for(int scoop:scoops){
+        auto position = global_pos3()+parts[scoop].precalc[0];
+        const char *sound_msgs[]={"Whirrrr","Ker-chunk","Swish","Cugugugugug","Horrifying"};
+        sounds::sound(position, (int) rng(20,35), sound_msgs[rng(0,3)]);
+        for(tripoint i:c){
```

Comment:
Okay. Do you mean that ```c``` should be called position? Position isn't a member in the vehicle class.